src/core/mac/mac.cpp

```diff
@@ -501,8 +501,10 @@ otError Mac::SendOutOfBandFrameRequest(otRadioFrame *aOobFrame)
 {
     otError error = OT_ERROR_NONE;
 
+    VerifyOrExit(aOobFrame != NULL, error = OT_ERROR_INVALID_ARGS);
     VerifyOrExit(mEnabled, error = OT_ERROR_INVALID_STATE);
```

Comment:
Shall we change the prototype to use reference instead of pointer?

Reply:
The C public ot API `otLinkOutOfBandTransmitRequest()` has a pointer `otRadioFrame *aOobFrame` param. A NULL check to ensure correct use of API somewhere would be good (either here in Mac or in `otLinkOutOfBandTransmitRequest()` implementation). I prefer passing pointer to this and perform the check here (also aligns well as `mOobFrame` being a pointer).

Reply:
Maybe it's time for us to make a code style rule about checking external arguments with `assert()` or `VerifyOrExit()`. In my opinion, `src/core/api` is the entry to OpenThread core, external illegal arguments can be guarded with `VerifyOrExit()`, while once an external argument is passed into internal components, it should be at least legal, so we should use `assert()` in internal components. With this rule if we get an assert error, we know there's a bug either in `src/core/api` or internal components. What do you think? @abtink @jwhui

Reply:
I am in favor of having less checks in general (to have smaller code overall). 
Added this NULL check due to debugging on some internal code using this API (possibly incorrectly).